airbyte-cdk/python/connector_builder/message_grouper.py

```diff
@@ -88,7 +88,7 @@ def _get_message_groups(
 
         while records_count < limit and (message := next(messages, None)):
             if self._need_to_close_page(at_least_one_page_in_group, message):
-                self._close_page(current_page_request, current_page_response, current_slice_pages, current_page_records)
+                self._close_page(current_page_request, current_page_response, current_slice_pages, current_page_records, True)
```

Comment:
I think this line challenges [this assumption](https://github.com/airbytehq/airbyte/pull/24173/commits/b4c45590ad05169767fa1e915ab00079c34b72d4#diff-01d26032991ab687a8a2142afa745bbb0b92e4cb190835a19c8f3b151a10c699R129). I honestly don't mind too much but it means that we expect he CDK to possibly continue the read on non breaking exceptions  without a response and we should make it clear in the behaviour of this method. I think it translate in removing the param `validate_page_complete` because it is now useless